src/auto/keymap.rs

```diff
@@ -0,0 +1,210 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gdk_sys;
+use glib::object::ObjectType as ObjectType_;
+use glib::signal::connect_raw;
+use glib::signal::SignalHandlerId;
+use glib::translate::*;
+use glib_sys;
+use pango;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::mem;
+use std::mem::transmute;
+use Display;
+use KeymapKey;
+use ModifierIntent;
+use ModifierType;
+
+glib_wrapper! {
+    pub struct Keymap(Object<gdk_sys::GdkKeymap, KeymapClass>);
+
+    match fn {
+        get_type => || gdk_sys::gdk_keymap_get_type(),
+    }
+}
+
+impl Keymap {
+    pub fn add_virtual_modifiers(&self, state: &mut ModifierType) {
+        unsafe {
+            gdk_sys::gdk_keymap_add_virtual_modifiers(self.to_glib_none().0, state.to_glib());
```

Comment:
This function sets bits in the `state`, but here you pass a copy of the `state` to the C function and don't read it back afterwards.

Reply:
Based on this code: https://code.woboq.org/gtk/gtk/gdk/x11/gdkkeys-x11.c.html#gdk_x11_keymap_add_virtual_modifiers, it updates the `state` so I'm not sure what you're saying. It seems good to me?

Reply:
```rust
    pub fn add_virtual_modifiers(&self, state: &mut ModifierType) {
        unsafe {
            let mut s = state.to_glib();
            gdk_sys::gdk_keymap_add_virtual_modifiers(self.to_glib_none().0, &mut s);
            *state = from_glib(state);
       }
    }
```

Reply:
Oh I see! Thanks!